roles/tas_single_node/tasks/certificates.yml

```diff
@@ -126,6 +143,17 @@
           -passin 'pass:{{ tas_single_node_fulcio_ca_passphrase }}'
           -out '{{ tas_single_node_remote_fulcio_root_ca }}'
         creates: "{{ tas_single_node_remote_fulcio_root_ca }}"
+      when: >
+        tas_single_node_fulcio_root_ca == ""
+        and (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_root_ca) | list | length) == 0
+        or tas_single_node_fulcio_private_key != ""
```

Comment:
Similar to one of the previous comments - if user first provided cert on their own and then they wanted to switch to role-generated cert, this wouldn't properly regenerate. I think the last line of the conditon should be replaced by checking a result of a new task which will verify whether the existing private/public key match the existing cert.